packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -212,6 +219,17 @@ const createManagerApi: CreateManagerApi = ({ onSubmit, clearOnUnmount, initiali
     }
   }
 
+  function reset(resetInitialValues?: AnyObject) {
+    state = {
+      ...state,
+      ...initialFormState(resetInitialValues || initialValues),
+      fieldListeners: state.fieldListeners,
+      registeredFields: state.registeredFields
+    };
+
+    state.registeredFields.forEach(resetFieldState);
```

Comment:
Does this trigger form re-render?

Reply:
Nothing triggers rerender now. I think we should focus on rerendering/optimization after we finish with all features.